packages/hdwallet-provider/src/constructor/getOptions.ts

```diff
@@ -2,22 +2,26 @@ import { MnemonicPhrase, PrivateKey } from "./types";
 import { ConstructorArguments } from "./ConstructorArguments";
 import * as Constructor from "./Constructor";
 import * as LegacyConstructor from "./LegacyConstructor";
+import { validateMnemonic } from "ethereum-cryptography/bip39";
+import { wordlist } from "ethereum-cryptography/bip39/wordlists/english";
 
 // check that the first argument is a mnemonic phrase
 const isMnemonicPhrase = (
   mnemonicPhraseOrPrivateKeys: MnemonicPhrase | PrivateKey[]
 ): mnemonicPhraseOrPrivateKeys is MnemonicPhrase =>
-  typeof mnemonicPhraseOrPrivateKeys === "string";
+  typeof mnemonicPhraseOrPrivateKeys === "string" &&
+  validateMnemonic(mnemonicPhraseOrPrivateKeys, wordlist);
 
-// check that the first argument is a list of private keys
+// check that the first argument is a list of private keys or a single key
 const isPrivateKeys = (
-  mnemonicPhraseOrPrivateKeys: MnemonicPhrase | PrivateKey[]
+  mnemonicPhraseOrPrivateKeys: PrivateKey | PrivateKey[]
```

Comment:
This should stay the same as the argument to `isMnemonicPhrase` (well, both should have all 3 options, really)... the idea of these functions are to take something that can be one of many things and tell the type-checker how it knows when it's figured out which one it is.